sftk/ingression.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from collections import defaultdict
+from sftk.email_handler import EmailHandler
+from sftk.common import KEYWORDS, KEYWORD_LOOKUP
+from sftk import log_config
+from sftk.utils import flatten_list
+
+class DataIngression:
+    """
+    Base class for Data Ingression processes
+    Empty for now
+    May in future provide functionality such as:
+    - Logging ingression processes and providing summaries
+
+    Could be standardised to provide input -> process -> output functionality
+    This would allow us to standardise the way data is ingressed into the system
+    """
+    def __init__(self):
+        pass
+
+    def run(self):
+        pass
+
+    def cleanup(self):
+        pass
+
+class EmailIngression(DataIngression):
+    """
+    JSON  data ingressed from emails
+    NOTE: Once an email is opened imap will mark it as read
+    This means if an email FAILS to be processed it will be marked as read and not picked up again
+    This is why summary logging is important for record keeping
+    The original email_to_sharepoint_copy.py script does not handle this
```

Comment:
we should probably rename this file, because the copy at the end really threw both of us off. Or maybe we won't need it.